mvn exec:exec launch flexo

bmangez opened the pull request:
Just fixing the exec modules so that mvn exec:exec launch Flexo (3 versions).

Commit:
fix a formating issue (formating eat a ')' )

Commit:
Now you can start flexo with 'mvn exec:exec' in any of the exec child module (except flexomaintainer).
It's not perfect : logging prefs, dock icon and stuff like that maybe wrong. But at least classpath is correct and all 3 versions are starting. (DID NOT TEST ON WINDOWS)

Commit:
Merge remote branch 'upstream/master'

Conflicts:
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/processeditor/gr/WKFNodeGR.java

gpolet commented:
Ok for this pull request, but couldn't it be improved by directly using a generic configuration using maven properties and by loading the adequate property file located in the flexobuild module.
Two questions come to my mind:
1) Does Maven allow to overide properties by loading them in their child but define a generic configuration in the parent module (exec)
2) I think that in any case, this would require to move the property file in each exec sub-module (flexobusiness, flexoentreprise, etc...), meaning that the build.xml (in the flexobuild module) should also be reworked to adapt to those changes.

The cherry on top of the cream, would be to also include some OS specific profiles for each OS. I am not sure that unknown VM args will be accepted by the JVM of alternate OS.

Anyway, thanks for the pull. I will look into it more deeply tomorrow and see if I can manage something. Feel free to go ahead on my propositions if you feel like it.

bmangez commented:
1) not sure it will work, but I can try it.
2) yes, it can be a good move
3) adding OS profile and debug profile can be a good move too.
4) You are right regarding VM args. I'm afraid it won't work on windows because the dock icon.

There is room for improvement ;-)

Commit:
Merge remote branch 'upstream/master'

Commit:
DocumentListener refactoring to reduce duplication

Commit:
exec plugin configuration improved :
- one single configuration of the exec plugin (only in the exec module)
- os specific execution args are defined in a profile activate regarding os-family (file : exec/pom.xml)
- profile 'debug' start flexo in debug mode allowing connection of a remote debugger (port 8001) (file : exec/pom.xml)
- each version of Flexo use the correct Flexo.properties
- assembly configuration restored
- classpath in flexobusinessplus import flexobusiness classpath and just specify additional artifacts (datamodeleditor)
- classpath in flexoentreprise import flexobusinessplus classpath and just specify additional artifacts (generators, ...)
- ResourceLocator modified to launch the correct Flexo.properties when running flexo with the maven-exec-plugin

bmangez commented:
A propos des improvements de la config du maven-exec-plugin.
Si vous acceptez le commit, le site (http://openflexo.org/developers/sources-and-build/) pourrait être mis à jour avec ceci :

An alternative to run flexo it to use the command '_mvn exec:exec_' under one of the module :
- openflexo/packaging/exec/flexobusiness
- openflexo/packaging/exec/flexobusinessplus
- openflexo/packaging/exec/flexoentreprise

You can use '_mvn exec:exec -Pdebug_' to start flexo in debug mode. Connect your remote debugger with the following configuration :
port : 8001
transport : socket
debugger mode : attach
host : localhost

Note that, when running with the debug profile, the application will wait the debugger on startup. You will see this log on your console :
Listening for transport dt_socket at address: 8001
If you don't need to start the debugger on startup, you change this behavior by specifying 'suspend=n' in the debugs.args in  exec/pom.xml

Commit:
Merge remote branch 'upstream/master'

Commit:
define test suite to force ordering of test methods

Commit:
Maven best practices :
* (MAJOR) define dependency version in dependencyManagement in the root pom.
* (MINOR) rename modules (not artifactid !) so that the name remember the parent module
* (MINOR) pom.xml layout harmonization

Commit:
ModuleLoader : remove single module load feature

Commit:
Maven best practices : remove duplicate dependency

Commit:
Huge ModuleLoader refactoring. see https://wiki.openflexo.com/display/FLXDEV/Work+in+progress

Commit:
Merge remote branch 'remotes/upstream/master'

Commit:
More integration tests + remove some useless code

Commit:
delete useless module (moved under flexointegrationtestsupport)

Commit:
Mechanism to avoid interaction between prefs used during integration tests and current user prefs.
Now : when running integration tests : a new Instance of the applicationDataDirectory is used (a temp directory) and so new preferences are used for each test. (applicationDataDirectory is created during test setup and clean up during tearDown).
(similar mechanism for FlexoResourceCenter was done in a previous commit, but not really explained in commit message)

I had to remove the old feature retrieving the "oldPrefs" from the "legacy" prefs location. Impact of this removal is really minor since the default pref location was changed a few years ago and so it may have a really small impact only on users who didn't use flexo during the last 3/4 years.

Commit:
Merge remote branch 'remotes/upstream/master'

Conflicts:
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/AdvancedPrefs.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/ApplicationData.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/Flexo.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/GeneralPreferences.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/application/FlexoApplicationAdapter.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/BugReportView.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/BugReportViewerWindow.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/NewBugReport.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/components/browser/ProjectElement.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/AutoSaveService.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoAutoSaveThread.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoModule.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoResourceCenterService.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/InteractiveFlexoResourceUpdateHandler.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/Module.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ModuleLoader.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ModuleLoadingException.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ProjectLoader.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/UserType.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/prefs/FlexoPreferences.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/rm/ResourceManagerModel.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/FlexoFrame.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/ModuleBar.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/ActionInitializer.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FlexoController.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FlexoServerInstanceManager.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FullInteractiveProjectLoadingHandler.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/WelcomePanelController.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/FileMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/HelpMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/ToolsMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/WindowMenu.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/FlexoModuleTestCase.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/AbstractGuiTest.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/ModuleLoaderTest.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/UserTypeTest.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestCG2.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestCGRepositoryDeletion.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestRoundTrip.java
	flexodesktop/generators/flexodocgenerator/src/main/java/org/openflexo/dg/html/DGJSGenerator.java
	flexodesktop/model/flexofoundation/pom.xml
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/gen/FlexoProcessImageBuilder.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/gen/ScreenshotGenerator.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/rm/FlexoGeneratedResource.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/FlexoProjectUtil.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/ProjectLoadingCancelledException.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/UnreadableProjectException.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/viewpoint/ExampleDrawingShema.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/viewpoint/ViewPointPalette.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/module/ModuleLoadingException.java
	flexodesktop/model/flexofoundation/src/test/java/org/openflexo/foundation/imported/TestImportedRoles.java
	flexodesktop/model/flexofoundation/src/test/java/org/openflexo/foundation/utils/FlexoProjectUtilTest.java
	flexodesktop/model/flexoinspector/pom.xml
	flexodesktop/model/flexopamela/pom.xml
	flexodesktop/model/flexoutils/pom.xml
	flexodesktop/model/flexoutils/src/test/java/org/openflexo/antar/expr/parser/TestJavaExpressionParser.java
	flexodesktop/model/flexoxmlcode/pom.xml
	flexodesktop/modules/flexocodegeneratormodule/src/main/java/org/openflexo/cgmodule/controller/action/OpenDMEntityInitializer.java
	flexodesktop/modules/flexodatamodeleditor/src/test/java/org/openflexo/dm/view/controller/TestDMController.java
	flexodesktop/modules/flexodocgeneratormodule/src/main/java/org/openflexo/dgmodule/controller/DGController.java
	flexodesktop/modules/flexodocgeneratormodule/src/test/java/org/openflexo/dg/test/TestPDF.java
	flexodesktop/modules/flexoscreeneditor/src/main/java/org/openflexo/ie/util/TriggerRepaintDocumentListener.java
	flexodesktop/modules/flexoscreeneditor/src/main/java/org/openflexo/ie/view/controller/action/AddComponentInitializer.java
	flexodesktop/modules/flexoscreeneditor/src/test/java/org/openflexo/ie/util/TriggerRepaintDocumentListenerTest.java
	flexodesktop/modules/flexoscreeneditor/src/test/java/org/openflexo/ie/view/controller/TestIEController.java
	flexodesktop/modules/flexosourcegeneratormodule/src/main/java/org/openflexo/sgmodule/controller/action/OpenDMEntityInitializer.java
	flexodesktop/modules/flexovieweditor/src/dev/java/org/openflexo/ve/fib/dialogs/CreateObjectPropertyDialogEDITOR.java
	flexodesktop/modules/flexovieweditor/src/test/java/org/openflexo/ontology/TestPizza.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/CalcLibraryViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/CalcViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/EditionPatternViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/main/java/org/openflexo/vpm/controller/CEDController.java
	flexodesktop/modules/flexoviewpointmodeler/src/main/java/org/openflexo/vpm/controller/CalcLibraryBrowser.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/controller/action/OpenOperationComponentInitializer.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/controller/action/SetAndOpenOperationComponentInitializer.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/view/ProcessBrowserView.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/view/WorkflowBrowserView.java
	flexodesktop/modules/flexoworkfloweditor/src/test/java/org/openflexo/wkf/view/controller/TestWKFController.java
	flexodesktop/technologymodules/bpslibtechnologymodule/pom.xml
	flexodesktop/technologymodules/javatechnologymodule/pom.xml
	flexodesktop/technologymodules/maventechnologymodule/pom.xml
	flexodesktop/technologymodules/postgrestechnologymodule/pom.xml
	flexodesktop/technologymodules/springtechnologymodule/pom.xml
	flexoserver/flexoexternalbuilders/src/main/java/org/openflexo/builders/FlexoExternalMain.java
	flexoserver/flexoexternalbuilders/src/main/java/org/openflexo/builders/FlexoExternalMainWithProject.java
	packaging/exec/flexobusiness/pom.xml
	packaging/exec/flexobusinessplus/pom.xml
	packaging/exec/flexoenterprise/pom.xml
	packaging/exec/pom.xml
	pom.xml
	testsupport/integration/flexoenterprisetest/src/test/java/org/openflexo/module/EnterpriseModuleTest.java

Commit:
Merge remote branch 'remotes/upstream/master'

Conflicts:
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/AdvancedPrefs.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/ApplicationData.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/Flexo.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/GeneralPreferences.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/application/FlexoApplicationAdapter.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/BugReportView.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/BugReportViewerWindow.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/br/view/NewBugReport.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/components/browser/ProjectElement.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/AutoSaveService.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoAutoSaveThread.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoModule.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/FlexoResourceCenterService.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/InteractiveFlexoResourceUpdateHandler.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/Module.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ModuleLoader.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ModuleLoadingException.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/ProjectLoader.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/module/UserType.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/prefs/FlexoPreferences.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/rm/ResourceManagerModel.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/FlexoFrame.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/ModuleBar.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/ActionInitializer.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FlexoController.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FlexoServerInstanceManager.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/FullInteractiveProjectLoadingHandler.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/controller/WelcomePanelController.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/FileMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/HelpMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/ToolsMenu.java
	flexodesktop/GUI/flexo/src/main/java/org/openflexo/view/menu/WindowMenu.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/FlexoModuleTestCase.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/AbstractGuiTest.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/ModuleLoaderTest.java
	flexodesktop/GUI/flexo/src/test/java/org/openflexo/module/UserTypeTest.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestCG2.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestCGRepositoryDeletion.java
	flexodesktop/generators/flexocodegenerator/src/test/java/org/openflexo/generator/TestRoundTrip.java
	flexodesktop/generators/flexodocgenerator/src/main/java/org/openflexo/dg/html/DGJSGenerator.java
	flexodesktop/model/flexofoundation/pom.xml
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/gen/FlexoProcessImageBuilder.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/gen/ScreenshotGenerator.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/rm/FlexoGeneratedResource.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/FlexoProjectUtil.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/ProjectLoadingCancelledException.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/utils/UnreadableProjectException.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/viewpoint/ExampleDrawingShema.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/foundation/viewpoint/ViewPointPalette.java
	flexodesktop/model/flexofoundation/src/main/java/org/openflexo/module/ModuleLoadingException.java
	flexodesktop/model/flexofoundation/src/test/java/org/openflexo/foundation/imported/TestImportedRoles.java
	flexodesktop/model/flexofoundation/src/test/java/org/openflexo/foundation/utils/FlexoProjectUtilTest.java
	flexodesktop/model/flexoinspector/pom.xml
	flexodesktop/model/flexopamela/pom.xml
	flexodesktop/model/flexoutils/pom.xml
	flexodesktop/model/flexoutils/src/test/java/org/openflexo/antar/expr/parser/TestJavaExpressionParser.java
	flexodesktop/model/flexoxmlcode/pom.xml
	flexodesktop/modules/flexocodegeneratormodule/src/main/java/org/openflexo/cgmodule/controller/action/OpenDMEntityInitializer.java
	flexodesktop/modules/flexodatamodeleditor/src/test/java/org/openflexo/dm/view/controller/TestDMController.java
	flexodesktop/modules/flexodocgeneratormodule/src/main/java/org/openflexo/dgmodule/controller/DGController.java
	flexodesktop/modules/flexodocgeneratormodule/src/test/java/org/openflexo/dg/test/TestPDF.java
	flexodesktop/modules/flexoscreeneditor/src/main/java/org/openflexo/ie/util/TriggerRepaintDocumentListener.java
	flexodesktop/modules/flexoscreeneditor/src/main/java/org/openflexo/ie/view/controller/action/AddComponentInitializer.java
	flexodesktop/modules/flexoscreeneditor/src/test/java/org/openflexo/ie/util/TriggerRepaintDocumentListenerTest.java
	flexodesktop/modules/flexoscreeneditor/src/test/java/org/openflexo/ie/view/controller/TestIEController.java
	flexodesktop/modules/flexosourcegeneratormodule/src/main/java/org/openflexo/sgmodule/controller/action/OpenDMEntityInitializer.java
	flexodesktop/modules/flexovieweditor/src/dev/java/org/openflexo/ve/fib/dialogs/CreateObjectPropertyDialogEDITOR.java
	flexodesktop/modules/flexovieweditor/src/test/java/org/openflexo/ontology/TestPizza.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/CalcLibraryViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/CalcViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/dev/java/org/openflexo/vpm/fib/EditionPatternViewEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/main/java/org/openflexo/vpm/controller/CEDController.java
	flexodesktop/modules/flexoviewpointmodeler/src/main/java/org/openflexo/vpm/controller/CalcLibraryBrowser.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/controller/action/OpenOperationComponentInitializer.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/controller/action/SetAndOpenOperationComponentInitializer.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/view/ProcessBrowserView.java
	flexodesktop/modules/flexoworkfloweditor/src/main/java/org/openflexo/wkf/view/WorkflowBrowserView.java
	flexodesktop/modules/flexoworkfloweditor/src/test/java/org/openflexo/wkf/view/controller/TestWKFController.java
	flexodesktop/technologymodules/bpslibtechnologymodule/pom.xml
	flexodesktop/technologymodules/javatechnologymodule/pom.xml
	flexodesktop/technologymodules/maventechnologymodule/pom.xml
	flexodesktop/technologymodules/postgrestechnologymodule/pom.xml
	flexodesktop/technologymodules/springtechnologymodule/pom.xml
	flexoserver/flexoexternalbuilders/src/main/java/org/openflexo/builders/FlexoExternalMain.java
	flexoserver/flexoexternalbuilders/src/main/java/org/openflexo/builders/FlexoExternalMainWithProject.java
	packaging/exec/flexobusiness/pom.xml
	packaging/exec/flexobusinessplus/pom.xml
	packaging/exec/flexoenterprise/pom.xml
	packaging/exec/pom.xml
	pom.xml
	testsupport/integration/flexoenterprisetest/src/test/java/org/openflexo/module/EnterpriseModuleTest.java

Commit:
Merge remote branch 'remotes/upstream/master'

Conflicts:
	flexodesktop/modules/flexovieweditor/src/dev/java/org/openflexo/ve/fib/dialogs/CreateObjectPropertyDialogEDITOR.java
	flexodesktop/modules/flexoviewpointmodeler/src/main/resources/Fib/EditionPatternView.fib
